rules/prefer-replace-all.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+function checkReplace(node) {
+	return node.callee.property.name === 'replace';
+}
+
+function checkGlobalFlag(node) {
+	const searchPattern = node.arguments[0];
+	return Object.prototype.hasOwnProperty.call(searchPattern, 'regex') && searchPattern.regex.flags === 'g';
+}
+
+function checkLiteralCharactersOnly(node) {
+	const searchPattern = node.arguments[0].regex.pattern;
+	const specialCharacters = searchPattern.match(/[$()*+.?[\\\]^{}]/g);
+	const specialCharactersCount = specialCharacters ? specialCharacters.length : 0;
+	const escapedSpecialCharacters = searchPattern.match(/\\[$()*+.?[\\\]^{}]/g);
+	const escapedSpecialCharactersCount = escapedSpecialCharacters ? escapedSpecialCharacters.length : 0;
+	return specialCharactersCount === 2 * escapedSpecialCharactersCount;
+}
+
+function replaceNode(node, fixer) {
+	const stringName = node.callee.object.name;
+	const searchPattern = node.arguments[0].regex.pattern.replace(/\\(.)/g, '$1');
```

Comment:
This line is not right, `/\\a/g` is searching for `\a` not `a`, maybe `.replace(/\\\\(.)/g, '\\$1')`?.

Also, funny thing.

I think we can replace it with `.replace(/\\\\/g, '\\')`, then this line will against this rule itself, so maybe `.replaceAll('\\\\', '\\')`?

Reply:
Hmm. On second thought, that replace shouldn't be necessary at all. However, I do need to add an escape for any double quotes in the original regex expression.

Yeah, that ocurred to me while I was writing it. Very ironic. I decided not to use replaceAll since it doesn't have support in node yet and it would be transpiled back to the expression I have if we ran it through babel.